src/viztracer/cellmagic.py

```diff
@@ -32,6 +36,40 @@ def view():  # pragma: no cover
 
             display(button)
 
+        def make_kwargs(self, line, tracer_signature, viewer_signature):
```

Comment:
You basically wrote an argument parser here, which would be difficult to maintain and requires a lot of tests (viztracer keeps a 100% coverage). I don't think this should be the way to go, especially considering there could be conflict arguments (for example, eliminating `--once` is a very bad idea for the viewer).

I don't want the magic cells to have full flexibility as the CLI, that could be unnecessarily difficult to maintain. However, having some options like ``--port`` might be beneficial. Still, I prefer a white list way instead of this auto-importing kind of way (this implementation will introduce a silent dependency that VizTracer and ServerThead can't share args with the same name, which is not ideal).

To be honest, cell magic is not the top-priority for viztracer - it would be nice if it works, but it's totally fine that it won't work under some circumstances. To leverage its full capability, the users can always use the inline `VizTracer` context and open the report with code.

So, I think I'd prefer this piece to be removed, and if you want some useful argument for the cell magic, use something like `IPython.core.magic_arguments`, which we do not need to test for its validity.